syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.arkai.inference.musev";
option java_outer_classname = "MuseVService";
option objc_class_prefix = "museV";
option go_package = "gitlab.in.promiseland.live/ark/ark-api/ai_grpc";

package musev;

service MuseV {
  rpc Generate(Img2VideoInputMsg) returns (Img2VideoResponse) {}
}

message Img2VideoInputMsg {
  string conditionImages = 1;         // oss源文件路径名（原图）
  string ipadapterImage = 2;         // oss源文件路径名（原图），默认空
  string referImage = 3;         // oss源文件路径名（原图），默认空
  string prompt = 4;
  float scaleRatio = 5;
  float eyeBlinksFactor = 6;
  string dest = 7;
  int32 fps = 8;
  int32 length = 9;
  float motionSpeed = 10;
  int32 batch = 11;
}

message Img2VideoResponse {
  bool result = 1;
  string dest = 2;
}
